src/model.rs

```diff
@@ -51,15 +35,18 @@ pub struct Model {
 
     // the view that the UI treats as the topmost grammar to start rendering from.
     view_root: Coordinate,
+    f_select_cell: Coordinate,
+    l_select_cell: Coordinate,
```

Comment:
consider defining these as `Option<Coordinate>` and initializing them to `None`

Reply:
also make the names here clearer: `first_selected_cell` and `last_selected_cell`